jOOL-java-8/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -979,4 +976,149 @@ else if (percentile == 1.0)
+     * }
+     * Besides, self defined class is also allowed with mapping function:
+     * {@code
+     * Seq.of(new Node(1), new Node(1), new Node(1), new Node(1)).collect(Agg.stddevBy(mapping1));
+     * }
+     * </pre>
+     *
+     * @param function mapping function from object to double value
+     * @return the stddev value
+     * @version 1.0
+     * @author Jichen Lu
+     * @date 2021-04-25
+     */
+    public static <T> Collector<T, ?, Optional<Double>> stddevBy(
+            Function<? super T,
+                    Double> function) {
+        return Collector.of(
+                (Supplier<ArrayList<T>>) ArrayList::new,
+                ArrayList::add,
+                (l1, l2) -> {
+                    l1.addAll(l2);
+                    return l1;
+                },
+                l -> {
+                    int size = l.size();
+                    if (size == 0) {
+                        return Optional.empty();
+                    }
+                    double variance = getVariance(function, l, size);
+                    double stddev;
+
+
+                    stddev = Math.sqrt(variance);
+
+                    return Optional.of(stddev);
+                }
+        );
+    }
+
+
+    //CS304 Issue link: https://github.com/jOOQ/jOOL/issues/360
+
+    /**
+     * Calculate the variance of the given object collectors,
+     * based on the mapping function from object to double number.
+     *
+     * <p><pre> Usage of aggregation function varianceBy():
+     * The mapping function is a function mapping the objects to a double value, for instance:
+     * {@code
+     * Function<Integer, Double> mapping = e -> Double.valueOf(e);
+     * }
+     * The specific usage of varianceBy is like:
+     * {@code
+     * Seq.of(1, 1, 1, 1).collect(Agg.varianceBy(mapping));
+     * }
+     * Besides, self defined class is also allowed with mapping function:
+     * {@code
+     * Seq.of(new Node(1), new Node(1), new Node(1), new Node(1)).collect(Agg.varianceBy(mapping1));
+     * }
+     * </pre>
+     *
+     * @param function mapping function from object to double value
+     * @return the stddev value
+     * @version 1.0
+     * @author Jichen Lu
+     * @date 2021-04-25
+     */
+    public static <T> Collector<T, ?, Optional<Double>> varianceBy(
+            Function<? super T,
+                    Double> function) {
+        return Collector.of(
+                (Supplier<ArrayList<T>>) ArrayList::new,
+                ArrayList::add,
+                (l1, l2) -> {
+                    l1.addAll(l2);
+                    return l1;
+                },
+                l -> {
+                    int size = l.size();
+                    if (size == 0) {
+                        return Optional.empty();
+                    }
+                    double variance = getVariance(function, l, size);
+                    return Optional.of(variance);
+                }
+        );
```

Comment:
This entire collector is unnecessary. You can just use `Collectors.collectingAndThen(toList(), varianceCalculationHere)`